skimage/feature/peak.py

```diff
@@ -123,6 +157,15 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
 
     out = np.zeros_like(image, dtype=np.bool)
 
+    # no peak for a trivial image
+    if np.all(image == image.flat[0]):
+        if indices is True:
+            return np.empty((0, 2), np.int)
+        else:
+            return out
+
+    threshold_abs = threshold_abs if threshold_abs is not None else image.min()
```

Comment:
Minor: I would move the block "if type(exclude_border)..." few lines above just before or after this line, to gather these lines which have the same spirit.

Reply:
sounds good.

Reply:
Only this one to fix, and it will be good for me.